### ISSUE TITLE
[Support]: Motion frames on frigate startup cause object detection to "stick" on all parked cars in the area. Driving high detector usage. 

### ISSUE THREAD

**Issue by aredon** [blakeblackshear/frigate#9651]
### Describe the problem you are having

Pretty much title. I upgraded to V13 and have been going through my cameras to update the configs and noticed that object detection is stuck on all the parked cars on my street that are in frame. As far as I can tell these just keep detecting until object detection is toggled off. 

From what I can tell there are some errant motion boxes being drawn when frigate first starts. These are driving object detection for most of the frame - and then it just stays on because the object hasn't left. 

### Version

0.13.1-34FB1C2

### Frigate config file

```yaml
#GLOBAL FFMPEG
ffmpeg:
  output_args:
    record: preset-record-generic-audio-aac

#GLOBAL OBJECT SETTINGS
#Objects list: https://docs.frigate.video/configuration/objects/
objects:
  track:
    - person
    - dog
    - cat
    - bird
    - car

snapshots:
  retain:
    default: 7

birdseye:
  enabled: True
  mode: motion

#CAMERAS
cameras:
  ##################################################### OFFICE
  office:
    ffmpeg:
      inputs:
        - path: rtsp://Frigate:REDACTED@192.168.1.201:554//cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
        - path: rtsp://Frigate:REDACTED@192.168.1.201:554//cam/realmonitor?channel=1&subtype=0
          roles:
            - record
    zones:
      porch:
        coordinates: 74,251,302,336,640,425,640,480,535,480,406,480,298,480,180,480,150,452,64,370,0,293,68,295
    detect:
      enabled: True
      # width: 640
      # height: 480
    record:
      enabled: True
      retain:
        days: 7
        mode: motion
      events:
        required_zones: 
          - porch
        retain:
          default: 7
          mode: active_objects
    snapshots:
      enabled: True
      bounding_box: True
      required_zones: 
        - porch
    motion:
      mask:
        - 211,86,640,36,640,480,573,480,491,480,482,420,465,364,469,292,375,271,285,254,277,325,246,328,181,300,66,262,71,137
        - 640,451,640,480,382,480,383,454
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Debian

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Misc Amcrest Devices

**Comment by NickM-27** [comment]
> As far as I can tell these just keep detecting until object detection is toggled off.

That's not how frigate works. Frigate knows when objects are stationary and doesn't run detection continuously (only periodically)

**Comment by NickM-27** [comment]
I'd suggest providing a screenshot of the system page

**Comment by aredon** [comment]
> > As far as I can tell these just keep detecting until object detection is toggled off.
> 
> That's not how frigate works. Frigate knows when objects are stationary and doesn't run detection continuously (only periodically)

Not sure what to tell you but that's what's happening. The whole debug frame is full of object bounding boxes with no motion boxes to speak of after the first couple frames. First few seconds of frames there's a red motion box over literally 95%+ of the frame, object bounding boxes show up right after that and stay highlighted in perpetuity with no motion boxes drawn on any subsequent frames for at least the ten minutes that I let it run.  I am currently brining up the motion threshold to see if that helps avoid the whole-frame motion box on startup - just sucks that I might not be able to go below whatever that ends up being.

**Comment by NickM-27** [comment]
> Not sure what to tell you but that's what's happening. The whole debug frame is full of object bounding boxes with no motion boxes to speak of after the first couple frames. 

yes, but that doesn't mean object detection is running continuously. Frigate tracks objects and knows they are there and only looks for objects if there is motion or on an interval. This is covered in the docs https://docs.frigate.video/configuration/stationary_objects

**Comment by aredon** [comment]
> > Not sure what to tell you but that's what's happening. The whole debug frame is full of object bounding boxes with no motion boxes to speak of after the first couple frames.
> 
> yes, but that doesn't mean object detection is running continuously. Frigate tracks objects and knows they are there and only looks for objects if there is motion or on an interval. This is covered in the docs https://docs.frigate.video/configuration/stationary_objects

I'm familiar with stationary objects. I would not expect bounding boxes to have shifting percentages, colors, and shapes if object detection is not running. I can also literally hear the cooling fan on my server chugging.

CPU detector is at 253% right now while this is going on. Normally this floats around 20%

![image](https://github.com/blakeblackshear/frigate/assets/1328757/7b708664-b0c2-4220-8569-37405feb2124)

![image](https://github.com/blakeblackshear/frigate/assets/1328757/6d755f69-7714-4152-a6d1-0cad0e43d210)

![image](https://github.com/blakeblackshear/frigate/assets/1328757/81c8e5e7-5033-4149-8db2-fb06784ab7db)

**Comment by aredon** [comment]
![image](https://github.com/blakeblackshear/frigate/assets/1328757/3f60711c-5282-4fd8-9d39-7e62515cc326)

**Comment by NickM-27** [comment]
most cameras are not running detection at all. Some have relatively low detect fps, but the high CPU usage is because you are using a cpu detector which is not recommended. You can watch the debug live view with regions enabled and that will show what detection is being done, detection would only be running constantly if there is motion or the bounding boxes are not consistent enough for the object to be considered stationary. 

You have a lot of skipped fps because the inference speed of your CPU is too high so that is likely causing important detections to be skipped and making the problem much worse

**Comment by aredon** [comment]
Most cameras are not pointing outside so don't see much action. It's front door and office that can see the street. I have been running for months with this CPU setup in my VM without any skipped frames (I've been monitoring very closely). It's only after the update I've had problems.

Here's how it looks once I have toggled detection off and then back on again:
![image](https://github.com/blakeblackshear/frigate/assets/1328757/221e1396-f574-4d3d-9e92-50bb4b4dedb2)

No detection is occurring because nothing is happening in frame. It's only after startup (or probably any motion that crosses the frame) that this skyrockets. 

I will try restarting the server again and paying attention to regions instead of bounding boxes.

**Comment by aredon** [comment]
Ok after a restart I can confirm there are large detection regions drawn over most of the frame. There is no red motion boxes happening.

**Comment by NickM-27** [comment]
right, on startup frigate runs detection to get an initial idea of what objects are present and did not just appear when motion is first detected.

by default an object box needs to be similar for 50 frames and then it will be considered stationary. If no motion is occurring and detection continues that is because the object box is not consistent. Or it can be because your CPU detector is causing skipped frames so the object is presumed missing

**Comment by aredon** [comment]
Perhaps I am not being clear. Here is the order of events after startup:
1.) Large red motion box covers 95%+ of the frame
2.) Object detection regions appear 
3.) Motion box goes away
4.) Detection regions persist for 10+ minutes

The issue stops if I set the motion threshold to 120 for this camera. Which seems a bit high? Do you know what the threshold was in v12?

It does seem likely that skipped frames are part of the problem. Maybe I need to crank up the motion threshold for all my cameras? I am seeing some persistent motion boxes on some of the other cameras. So that may actually be the bulk of the CPU usage.

**Comment by NickM-27** [comment]
I don't understand. You say:

> 3.) Motion box goes away
> 4.) Detection regions persist for 10+ minutes

but then later say:

> I am seeing some persistent motion boxes on some of the other cameras.

so does motion stop or does it keep going?

if motion is ongoing when nothing is changing then that means motion is too sensitive.

**Comment by aredon** [comment]
Apologies, there are two sets of cameras.
Group 1: Outdoor cameras with a persistent object detection issue. No motion detected.
Group 2: Indoor cameras with persistent motion detection at startup. This appears to taper down over a few minutes and then stop. It's unclear to me if that means the motion threshold needs dialed up or not.

**Comment by aredon** [comment]
Ok. Motion has settled on all the other cameras after being started for about 5 minutes, but, something moved on the front_door camera and now the detection region is stuck on as before. 

![image](https://github.com/blakeblackshear/frigate/assets/1328757/2c9dc9bf-4b81-4fd8-9a1c-afa61b0feb2c)

**Comment by NickM-27** [comment]
Might be good to see a screen recording of the debug view while this is happening so I can understand what is going on exactly

**Comment by aredon** [comment]
I can absolutely arrange that. Give me a minute.

**Comment by aredon** [comment]
This should be pretty good.
https://github.com/blakeblackshear/frigate/assets/1328757/357a6582-4f46-4283-ac5e-25fc6eb3825f

**Comment by NickM-27** [comment]
There is no motion going on here, the problem is that the cars are not detected consistently so detection continues to be run. If you don't care about those cars being detected could use a min_area filter for cars.

**Comment by aredon** [comment]
What's causing the cars to not detect consistently?

**Comment by NickM-27** [comment]
the default model isn't trained on security camera images, we can see from the video that the cars are detected with a very low score. also some of them look like false positives and they are not actually cars

**Comment by aredon** [comment]
So the model is kind of alternating between detecting a car and not detecting a car and thus persists at trying to detect in that area? I'll give the min_area filter a try.

**Comment by kzgrey** [comment]
I believe I have this same problem.  The end result is that I have one video recording in progress constantly.  Disabling the Car detection and re-enabling it appears to fix the problem.  Turning Car detection back on makes the problem occur again.  Right now, on my Events tab, I see two "in progress" recordings for the same camera.

**Comment by NickM-27** [comment]
An in progress event does not mean it is being recorded, depends on the config. There is also a docs section for handling parked cars
